src/algebra/regular/smul.lean

```diff
@@ -173,16 +173,30 @@ end comm_monoid
 
 end is_smul_regular
 
+section group
+
+variables {G : Type*} [group G]
+
+/-- An element of a group acting on a monoid is regular. This relies on the availability
+of the inverse given by groups, since there is no `left_cancel_smul` typeclass. -/
+lemma is_smul_regular_of_group [mul_action G R] (g : G) : is_smul_regular R g :=
+begin
+  intros x y h,
+  convert congr_arg ((•) g⁻¹) h using 1;
+  simp [←smul_assoc]
+end
+
+end group
+
 variables [monoid_with_zero R] [has_zero M] [mul_action_with_zero R M]
 
 /-- Any element in `units R` is `M`-regular. -/
-lemma units.is_smul_regular (a : units R) : is_smul_regular M (a : R)
-:=
+lemma units.is_smul_regular_coe (a : units R) : is_smul_regular M (a : R) :=
 is_smul_regular.of_mul_eq_one a.inv_val
```

Comment:
I think this name was probably better as it was; for instance, `units.mul_inv` is about `(↑a) * ↑a⁻¹ = 1` not `a * a⁻¹ = 1`, but does not have `coe` in its name. I think this is true for most of the units API.